src/library_syscall.js

```diff
@@ -1953,8 +1997,12 @@ for (var x in SyscallsLibrary) {
   if (!SyscallsLibrary[x + '__deps']) SyscallsLibrary[x + '__deps'] = [];
   SyscallsLibrary[x + '__deps'].push('$SYSCALLS');
 #if USE_PTHREADS
-  // proxy all syscalls synchronously, for their return values
-  SyscallsLibrary[x + '__proxy'] = 'sync';
+  // Proxy all syscalls synchronously, for their return values, except for ones
```

Comment:
I don't quite understand this sentence.
Actually I don't quite understand the one it's replacing either. Does it mean we are proxying synchronously in order to get the return values from the proxied call?

Reply:
Yes - if it's async proxied then we don't have a value to return to the caller. With sync proxying we block until the return value is ready, and return it.

I'll reword.